test/e2e/app.spec.ts

```diff
@@ -0,0 +1,44 @@
+import { afterAll, beforeAll, describe, expect, test } from 'vitest'
+import { preview } from 'vite'
+import type { PreviewServer } from 'vite'
+import puppeteer from 'puppeteer'
+import type { Browser, Page } from 'puppeteer'
+
+import { sleep } from '../utils'
+
+const port = 3142 // Unlike in life, where there is no such thing as free PI, servers usually don't run on it
+const BASE_URL = `http://localhost:${port}`
+const PAGES = {
+  login: `${BASE_URL}/login`,
+  createAccount: `${BASE_URL}/create-account`
+}
+
+describe("app.tsx", () => {
+  let server: PreviewServer
+  let browser: Browser
+  let page: Page
+
+  beforeAll(async () => {
+    server = await preview({ preview: { port } })
+    browser = await puppeteer.launch();
+    page = await browser.newPage();
+  });
+
+  afterAll(async () => {
+    await browser.close()
+    await new Promise((resolve) => server.httpServer.close(resolve))
+  });
+
+  test("should navigate to the create-account page", async () => {
+    await page.goto(PAGES.login);
+    const createAccountBtn = await page.waitForSelector("#create-account-btn");
```

Comment:
Can we replace this with something like ` await page.waitForSelector("a[href='/create-account]'");`?

I guess it could match other links instead of the button, but I'm not a fan of adding ids just for testing.

Reply:
Sure, thats a good suggestion. I actually wanted to use dedicated `data-test-id` originally but scrapped that idea. I'll refactor accordingly, thanks :)